src/maicos/lib/math.py

```diff
@@ -795,8 +805,16 @@ def compute_rdf_structure_factor(
     struct_factor : numpy.ndarray
         structure factor
 
+    Raises
+    ------
+    ValueError
+        If the distance array ``r`` is not equally spaced.
     """
     dr = r[1] - r[0]
+
+    if (abs(r[1:] - r[:-1] - dr) > 1e-6).any():
```

Comment:
I like this representation more
```suggestion
    if (abs(np.diff(r) - dr) > 1e-6).any():
```